lib/spack/spack/build_systems/intel.py

```diff
@@ -1309,3 +1318,48 @@ def uninstall_ism(self):
 
     # Check that self.prefix is there after installation
     run_after('install')(PackageBase.sanity_check_prefix)
+
+    # IntelPackages have their own test methods that collate the relevant tests
+    # Tests for each component are here in IntelPackage
+    def test_compiler(self):
+        # Get compiler tests
+        compiler_test_dir = self.test_dir.join('compiler_tests')
+        shutil.copytree(os.path.join(spack.paths.tests_path, 'compilers'),
+                        compiler_test_dir)
+
+        # compile and run each test
+        langs = ('c', 'c++', 'fortran')
+        compilers = ('icc', 'icpc', 'ifort')
+        for lang, compiler in zip(langs, compilers):
+            lang_dir = os.path.join(compiler_test_dir, lang)
+            for test in os.listdir(lang_dir):
+                full_path = os.path.join(compiler_test_dir, test)
+                exe_name = '%s.exe' % test
+
+                # Get the right compiler for this test
+                compiler = test_compiler_for_source(test)
+
+                ###
+                # Compile the test
+                ###
+                # once we reconcile compilers and dependencies this should
+                # query package for appropriate cxx11 flag
+                # only the hello_cxx11 test requires c++11
```

Comment:
Eventually it would be part of the contract for the `cxx` virtual package.

Before we implement contracts for virtuals, it would be defined individually on the compiler packages, the way it's currently defined in `lib/spack/spack/compilers/*.py`